src/main/java/View/MainStage.java

```diff
@@ -130,33 +136,48 @@ public void updateView(String FENPosition) {
      * 
      * @param possibleMoves
      */
-    public void addMoveOptionsCircles(ArrayList<PositionView> possibleMoves) {
+    public void addMoveOptionsCircles(PositionView positionView , ArrayList<PositionView> possibleMoves) {
+        boardView.getSquareView(positionView).changeSquareColour(SquareColour.CLICKED);
         for (PositionView position:possibleMoves) {
-            SquareView squareView = boardSquaresView.getSquareView(position);
+            SquareNode squareView = boardView.getSquareView(position);
             squareView.addCircle();
             circlesActivated.add(squareView);
         }
     }
 
+    /**
+     * Removing the circles of the possible moves and return the clicked square back to its normal colour
+     */
+    public void removeMoveOptionsCircles(PositionView positionView) {
+        boardView.getSquareView(positionView).changeSquareColour();
+        for (SquareNode squareview:this.circlesActivated) {
+            squareview.removeCircle();
+        }
+    }
+
     /**
      * Removing the circles of the possible moves
      */
     public void removeMoveOptionsCircles() {
-        for (SquareView squareview:this.circlesActivated) {
+        for (SquareNode squareview:this.circlesActivated) {
             squareview.removeCircle();
         }
     }
 
     public void promotionPopup(boolean isWhite){
-        PromotionView promotionView = new PromotionView(listener, isWhite);
+        PromotionModal promotionView = new PromotionModal(listener, isWhite);
         promotionView.show();
     }
 
     public void gameOverPopup(boolean isStaleMate, boolean isWhite){
-        new GameOverView(isStaleMate,isWhite);
+        GameOverModal gameOverModal = new GameOverModal(isStaleMate,isWhite);
+        gameOverModal.show();
+        if (gameOverModal.isNewGame()){
+            showMainMenu();
+        }
```

Comment:
is this a way of handling the user selection to play again..?

Reply:
Yeah. Might not be ideal but didn't want to go though the `Controller` if it's just updating the `View` again